crucible-syntax/src/Lang/Crucible/Syntax/Concrete.hs

```diff
@@ -1563,22 +1563,26 @@ atomSetter (AtomName anText) =
          return $ Some anAtom
 
     fresh =
-      do t <- reading (unary Fresh isType)
-         describe "user symbol" $
-           case userSymbol (T.unpack anText) of
-             Left err -> describe (T.pack (show err)) empty
-             Right nm ->
-               do loc <- position
-                  case t of
-                    Some (FloatRepr fi) ->
-                      Some <$>
-                        freshAtom loc (FreshFloat fi (Just nm))
-                    Some NatRepr ->
-                      Some <$> freshAtom loc (FreshNat (Just nm))
-                    Some tp
-                      | AsBaseType bt <- asBaseType tp ->
-                          Some <$> freshAtom loc (FreshConstant bt (Just nm))
-                      | otherwise -> describe "atomic type" $ empty
+      describe "user symbol" $
+      do -- Note that we are using safeSymbol below to create a What4 symbol
+         -- name, which will Z-encode names that aren't legal solver names. This
+         -- includes names that include hyphens, which are very common in
+         -- S-expression syntax. This is fine to do, since the Z-encoded name
+         -- name is only used for solver purposes; the original, unencoded name
+         -- is recorded separately.
+         let nm = safeSymbol (T.unpack anText)
+         t <- reading (unary Fresh isType)
+         loc <- position
+         case t of
+           Some (FloatRepr fi) ->
+             Some <$>
+               freshAtom loc (FreshFloat fi (Just nm))
+           Some NatRepr ->
+             Some <$> freshAtom loc (FreshNat (Just nm))
+           Some tp
+             | AsBaseType bt <- asBaseType tp ->
+                 Some <$> freshAtom loc (FreshConstant bt (Just nm))
+             | otherwise -> describe "atomic type" $ empty
```

Comment:
I don't think you want two layers of describe here...

Reply:
...although looking more carefully I notice this isn't new behavior. Might want to fix it up anyway.

Reply:
Good point. I think calling `describe` twice made more sense back when this function had two separate ways to fail, with one of the failures being `userSymbol`-related. Now that the only way to fail is by constructing a non-atomic type, I think that should be the only place we call `describe`. I've pushed a new version which does this.